torchao/prototype/quantized_training/int8_mixed_precision.py

```diff
@@ -0,0 +1,247 @@
+
+    def extra_repr(self):
+        return f"{super().extra_repr()}, config={self.config}"
+
+    @classmethod
+    def convert_linear(cls, module: nn.Module, config: Int8MixedPrecisionConfig = _DEFAULT_CONFIG):
+        if module.__class__ is nn.Linear:  # exact match, don't swap nn.Linear subclasses
+            module.__class__ = cls
+            module.config = config
+            return
+        for child in module.children():
+            cls.convert_linear(child, config)
+
+
+def _dynamic_int8_linear(input: Tensor, weight: Tensor) -> Tensor:
+    input_i8, input_scale = quantize_int8_rowwise(input)
+    weight_i8, weight_scale = quantize_int8_rowwise(weight)
+    return int8_mm_dequant(input_i8, weight_i8.T, input_scale, weight_scale)
+
+
+class _Int8MixedPrecisionLinear(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, input: Tensor, weight: Tensor, bias: Optional[Tensor], config: Int8MixedPrecisionConfig):
+        ctx.config = config
+        ctx.save_for_backward(input, weight)
+        ctx.bias = bias is not None
+
+        if ctx.config.output:
+            batch_dims = input.shape[:-1]
+            input = input.view(-1, weight.shape[1])
+            out = _dynamic_int8_linear(input, weight)
+            out = out.view(*batch_dims, weight.shape[0])
+        else:
+            out = input @ weight.T
+
+        out = out + bias if bias is not None else out
+        return out
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        input, weight = ctx.saved_tensors
+        grad_input = grad_weight = grad_bias = grad_config = None
+
+        batch_dims = grad_output.shape[:-1]
+        grad_output = grad_output.view(-1, weight.shape[0])
+        input = input.view(-1, weight.shape[1])
+
+        if ctx.needs_input_grad[0]:
+            if ctx.config.grad_input:
+                grad_input = _dynamic_int8_linear(grad_output, weight.T)
+            else:
+                grad_input = grad_output @ weight
+            grad_input = grad_input.view(*batch_dims, weight.shape[1])
+
+        if ctx.needs_input_grad[1]:
+            if ctx.config.grad_weight:
+                # TODO: check if transpose+quantize are fused
+                # grad_weight = _dynamic_int8_linear(grad_output.T, input.T)
+                grad_weight = _dynamic_int8_linear(input.T, grad_output.T).T  # this is slightly faster
+            else:
+                grad_weight = grad_output.T @ input
+
+        if ctx.needs_input_grad[2] and ctx.bias:
+            grad_bias = grad_output.sum(0)
+
+        return grad_input, grad_weight, grad_bias, grad_config
+
+
+def int8_mixed_precision_training(config: Int8MixedPrecisionConfig = _DEFAULT_CONFIG):
+    # TODO: right now `_get_linear_subclass_inserter()` will always set `requires_grad=False`
+    # when we have this out of prototype (or there are stable trainable tensor subclasses),
+    # update `_get_linear_subclass_inserter()` to allow `requires_grad=True`.
```

Comment:
If we need to do this in multiple places then maybe it's better to just add the flag :). Feel free to do it in a future PR